plugin/src/main/java/org/jenkinsci/plugins/structs/describable/UninstantiatedDescribable.java

```diff
@@ -0,0 +1,218 @@
+        this.klass = klass;
+        this.arguments = arguments;
+    }
+
+    public UninstantiatedDescribable(Map<String, ?> arguments) {
+        this(null,null,arguments);
+    }
+
+    /**
+     * If this nested describable has a suitable {@linkplain Symbol symbol name},
+     * this method returns one.
+     *
+     * <p>
+     * Either this or {@link #getKlass()} has to return a non-null value.
+     */
+    public @Nullable String getSymbol() {
+        return symbol;
+    }
+
+    public void setSymbol(String symbol) {
+        this.symbol = symbol;
+    }
+
+    /**
+     * {@code $class} is an alternative means to specify the class in case there's no symbol.
+     * Can be a short name if it's contextually unambiguous, or a FQCN.
+     *
+     * <p>
+     * Either this or {@link #getSymbol()} has to return a non-null value.
+     */
+    public @Nullable String getKlass() {
+        return klass;
+    }
+
+    public void setKlass(String klass) {
+        this.klass = klass;
+    }
+
+    /**
+     * Returns the model associated with this object.
+     *
+     * If this object was created from a model (via {@link #from(Object)}) this method
+     * returns that model.
+     */
+    public DescribableModel getModel() {
+        return model;
+    }
+
+    public void setModel(DescribableModel model) {
+        this.model = model;
+    }
+
+    /**
+     * All the nested arguments to this object.
+     */
+    public Map<String, ?> getArguments() {
+        return arguments;
+    }
+
+    /**
+     * Returns true if and only if the arguments is one and that is the only required parameter
+     * from the model.
+     *
+     * <p>
+     * This usually signals a short-hand syntax to write down the instantiation syntax.
+     */
+    public boolean hasSoleRequiredArgument() {
+        if (arguments.size()!=1)    return false;
+        if (model==null)    return false;
+
+        DescribableParameter p = model.getSoleRequiredParameter();
+        if (p==null)        return false;
+        return arguments.containsKey(p.getName());
+    }
+
+    /**
+     * For legacy use, we need to blow up this into a map form.
+     * This requires recursively blowing up any nested {@link UninstantiatedDescribable}s.
+     */
+    public Map<String,Object> toMap() {
+        Map<String,Object> r = new TreeMap<String, Object>();
+        for (Entry<String,?> e : arguments.entrySet()) {
+            Object v = e.getValue();
+            // see DescribableParameter.uncoerce for possible variety
+            v = toMap(v);
+            if (v instanceof List) {
+                ListIterator litr = ((List) v).listIterator();
+                while (litr.hasNext()) {
+                    litr.set(toMap(litr.next()));
+                }
+            }
+            r.put(e.getKey(),v);
+        }
+        if (klass !=null)
+            r.put(DescribableModel.CLAZZ, klass);
+        if (symbol!=null)
+            r.put(DescribableModel.SYMBOL,symbol);
```

Comment:
Is `SYMBOL` retrieved anywhere? This seems to be the only usage unless I missed something.